src/Discovery/src/ClientBase/DiscoveryHostBuilderExtensions.cs

```diff
@@ -0,0 +1,35 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the Apache 2.0 License.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Steeltoe.Connector;
+using System;
+using System.Reflection;
+
+namespace Steeltoe.Discovery.Client
+{
+    public static class DiscoveryHostBuilderExtensions
+    {
+        /// <summary>
+        /// Adds service discovery to your application. This method can be used in place of configuration via your Startup class.<para />
+        /// If <paramref name="optionsAction"/> is not provided, a <see cref="NoOpDiscoveryClient"/> will be configured
+        /// </summary>
+        /// <param name="hostBuilder">Your HostBuilder</param>
+        /// <param name="optionsAction">Select the discovery client implementation</param>
+        /// <remarks>Also configures named HttpClients "DiscoveryRandom" and "DiscoveryRoundRobin" for automatic injection</remarks>
+        /// <exception cref="AmbiguousMatchException">Thrown if multiple IDiscoveryClient implementations are configured</exception>
+        /// <exception cref="ConnectorException">Thrown if no service info with expected name or type are found or when multiple service infos are found and a single was expected</exception>
+        public static IHostBuilder AddServiceDiscovery(this IHostBuilder hostBuilder, Action<DiscoveryClientBuilder> optionsAction)
+        {
+            return hostBuilder.ConfigureServices((context, collection) => AddServices(collection, optionsAction));
+        }
+
+        private static void AddServices(IServiceCollection collection, Action<DiscoveryClientBuilder> optionsAction)
+        {
+            collection.AddServiceDiscovery(optionsAction);
+            collection.AddHostedService(services => new DiscoveryClientService(services.GetRequiredService<IDiscoveryLifecycle>()));
```

Comment:
I'm not a huge fan of how this is working... We don't currently need `DiscoveryClientService` for anything other than a mechanism to make sure `IDiscoveryLifeCycle` gets loaded up